mutation-testing/pr-differences/action.yml

```diff
@@ -0,0 +1,173 @@
+name: PR Differences Mutants
+
+inputs:
+  shard:
+    description: "The number of the shard to run (`-1` if ran without shards)"
+    required: false
+    default: -1
+  package-dimension:
+    description: "The dimension of the package. `big` for `stacks-node` and `stackslib`, `small` for others"
+    required: true
+
+runs:
+  using: "composite"
+
+  steps:
+    # Cleanup Runner
+    - name: Cleanup Runner
+      id: runner_cleanup
+      uses: stacks-network/actions/cleanup/disk@main
+
+    # Checkout the stacks-core code
+    - name: Checkout stacks-core repo
+      id: git_checkout_stacks_core
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        fetch-depth: 0
+
+    - name: Relative diff
+      shell: bash
+      run: |
+        git diff origin/${{ github.base_ref }}.. > git.diff
+
+    # Checkout the actions code
+    - name: Checkout actions repo
+      id: git_checkout_actions
+      uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
+      with:
+        repository: stacks-network/actions
+        ref: feat/mutation-testing
+        path: ./actions-repo
+
+    - name: Copy git diff
+      shell: bash
+      run: |
+        if [ ! -s git.diff ]; then
+          echo "Diff file missing!"
+        fi
+
+        mv git.diff ./actions-repo/mutation-testing/shell-scripts/
+
+    - uses: taiki-e/install-action@ac89944b5b150d78567ab6c02badfbe48b0b55aa # v2.20.16
+      name: Install cargo-mutants
+      with:
+        tool: cargo-mutants@23.12.2 # v23.12.2
+
+    - name: Update git diff
+      shell: bash
+      run: ./remove-deleted-file-lines.sh
+      working-directory: actions-repo/mutation-testing/shell-scripts
+
+    - name: Split diffs
+      shell: bash
+      run: |
+        # Check that the file exists before performing actions on it
+        if [ ! -s git.diff ]; then
+          echo "Diff file missing!"
+        fi
+
+        # Make a file containing all the mutants for the differences in the PR and a folder to split them into big and small packages
+        cargo mutants --in-diff git.diff --list > all_mutants.txt
```

Comment:
ambivalent here, but i think you should add a return code check to verity the command ran successfully. 
2 ways to do it mainly:
1. 
```
if [ $? -ne 0 ]; then 
  echo "error"
  exit $?
fi
```
2. use a method like this, to wrap the return in a variable: https://github.com/actions-rust-lang/rustfmt/blob/main/action.yml#L23